StratisCore.UI/src/app/shared/coins/coins.component.ts

```diff
@@ -0,0 +1,43 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { GlobalService } from '../services/global.service';
+import { BaseUnit } from "../BaseUnit";
+import { Subject } from 'rxjs';
+
+@Component({
+  selector: 'coins',
+  template: `
+    <span>{{ (amount ? amount : 0) | coinNotation:baseUnit.multiple | number:format }}</span><span *ngIf="showUnit"> {{ this.coinUnit | prefixCoinUnit:baseUnit.name }}</span>
+  `
+})
+export class CoinsComponent implements OnInit {
+
+  @Input()
+  amount: number = 0;
+
+  @Input()
+  showUnit: boolean = true;
+
+  @Input()
+  format: string = undefined;
+
+  baseUnit: BaseUnit;
+  coinUnit: string;
+  baseUnitSubscription: any;
+
+  constructor(private globalService: GlobalService) {
+    this.baseUnitSubscription = this.globalService.baseUnit.subscribe(b => {
+      this.baseUnit = b;
+      this.format = this.format != undefined ? this.format : b.defaultFormat;
```

Comment:
To be safe, I would go with `!!this.format ? this.format : b.defaultFormat` instead of `this.format != undefined ? this.format : b.defaultFormat`